.github/workflows/tests-01-main.yml

```
@@ -51,7 +51,16 @@ jobs:
                 echo "Failed to detect ${APP} as installed" >&2
                 exit 1
               else
-                ./deb-get purge "${APP}"
+                PUBLISHED_VER=$(./deb-get show "${APP}" | grep "Published:" | grep -oE '[^[:space:]]+$' || :)
+                INSTALLED_VER=$(dpkg-query -Wf '${Version}' "${APP}" || :)
+                if [ ! -z "${PUBLISHED_VER// /}" ] && [ "${PUBLISHED_VER}" != "${INSTALLED_VER}" ]; then
```

I think we should use the same `dpkg` test that deb-get uses for update rather than an exact string comparison here. 
```
if [ "${STATUS}" == "install ok installed" ] && dpkg --compare-versions "${VERSION_PUBLISHED}" gt "${VERSION_INSTALLED}"; then
``` 
so 
```suggestion
                if [ -n "${PUBLISHED_VER// /}" ] && dpkg --compare-versions "${PUBLISHED_VER}" gt "${INSTALLED_VER}" ; then
```

---

Wouldn't `ne` be preferable over `gt` in this instance? I'd think it would potentially catch more problems.

---

Actually I think not: it depends on what is a "problem".  I see the "problem" as being that the presented installed version evaluates lower than the presented available version causing persistent "update available". I think it is possible that `ne` would catch false-positives and generate problems where there are none (due to upstreams not following `man deb-versions` or making mistakes).  I think we need to allow for a graceful handling of "close enough" upstream versioning by just using the precise comparison that `deb-get` uses to identify an update.

---

My thinking was that it could also catch the theoretical opposite scenario, in which the installed version always evaluates higher, and updates never occur. I wonder if something like this might have less false positives than `ne`:
```
if [ -n "${PUBLISHED_VER// /}" ] && dpkg --compare-versions "${PUBLISHED_VER}" gt "${INSTALLED_VER}"  || dpkg --compare-versions "${PUBLISHED_VER}" lt "${INSTALLED_VER}"; then`
```
Or perhaps I'm just overthinking and this unlikely scenario isn't even worth worrying about.

---

I think we're both overthinking a little, but I'd also suggest that the "opposite scenario" is only going to occur if upstream has a problem to fix in their packaging/release processes.

---

Yeah, that makes sense.